### PR TITLE
#1090 customer invoice new data table/refactor start

### PR DESCRIPTION
**Pull request by Chris-Petty**
Works on #1090 but doesn't finish customer invoice page 😉 

## Change summary

Implements a basic version of:
- Data table and components
- Header row (no sorting)
- Reducer state management (editing totalQuantity, cell focus)
- Renderer call backs
- Hooks usage for the page rather than class api

### CODE REVIEW

File: src/widgets/DataTable/Row.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+const defaultStyles = StyleSheet.create({
+  row: {
+    flex: 1,
+    flexDirection: 'row',
+  },
+});
+
+export default React.memo(Row);
+
+//   (prevProps, nextProps) => {
```
Comment (joshxg) [inline src/widgets/DataTable/Row.js]: Can remove this comment block?

File: src/widgets/DataTable/Row.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable react/forbid-prop-types */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { View, StyleSheet } from 'react-native';
+
+const Row = ({ rowData, rowState, rowKey, renderCells }) => {
+  console.log('=================================');
```
Comment (joshxg) [inline src/widgets/DataTable/Row.js]: Should we get rid of all the logs in the components or did you want to keep them?

File: src/widgets/DataTable/HeaderRow.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable react/forbid-prop-types */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { View, StyleSheet } from 'react-native';
+
+const HeaderRow = React.memo(({ renderCells }) => (
```
Comment (joshxg) [inline src/widgets/DataTable/HeaderRow.js]: Should we do doc strings for all these components now? Dunno about making an issue for that or probably forget?

File: src/widgets/DataTable/Row.js (hunk 2 above)
Comment (Chris-Petty) [reply src/widgets/DataTable/Row.js]: Nah, I think keep them until we're happy with our implementation of customerInvoicePage

File: src/widgets/DataTable/Row.js (hunk 1 above)
Comment (Chris-Petty) [reply src/widgets/DataTable/Row.js]: Yup. I'll put the memo around the component, too.

File: src/widgets/DataTable/HeaderRow.js (hunk 3 above)
Comment (Chris-Petty) [reply src/widgets/DataTable/HeaderRow.js]: Do now I think

**Comment by joshxg** [comment]
I just added a doc string for `HeaderCell` https://github.com/openmsupply/mobile/pull/1108/commits/1897d796c8952a4fdb2275c0f2760a4f3c187729

**Comment by joshxg** [comment]
Changed the base to #1043-data-table-rewrite from develop

**Comment by Chris-Petty** [comment]
> I just added a doc string for `HeaderCell` [1897d79](https://github.com/openmsupply/mobile/commit/1897d796c8952a4fdb2275c0f2760a4f3c187729)

Ta I ran outa time yesterday 😢

**Pull request merged**